src/components/Anchor/index.js

```diff
@@ -21,13 +21,14 @@ const prefixByType = {
 const Anchor = ({
   url,
   children,
+  className = '',
   isSeparateTab = false,
   type = 'inline',
 }) => (
   <a
     href={url}
     target={isSeparateTab ? '_blank' : '_self'}
-    className={classNameByType[type]}
+    className={`${classNameByType[type]} ${className}`}
```

Comment:
Abuse example:

`<Anchor className='f6 link dim br1 ba ph3 pv2 mb2 dib black' />`
Now anchor looks like a Button, for which there is already a Button component.

So instead of this:
```
<Anchor url={lesson.edit_lesson_http_url} className='ma1'>
  <Button color='blue' size='extra-small'>
    {hasVideo ? editLessonTitleText : `${editLessonTitleText}`}
  </Button>
</Anchor>
```

I'm suggesting this:
```
<span className='ma1'>
  <Anchor url={lesson.edit_lesson_http_url}>
    <Button color='blue' size='extra-small'>
      {hasVideo ? editLessonTitleText : `${editLessonTitleText}`}
    </Button>
  </Anchor>
</span>
```

The only difference is the layout piece is wrapped instead of injected directly. This removes the layout implementation details so the component output is predictable and has a single responsibility; screens add wrapping layout generic elements and components render the same way given the same props.

This is the difference in functions:

```
const add = (x, y) => x + y
console.log(absolute(add(1, 1)))
// 20
```
vs
```
const add = (x, y, anotherFunction) => anotherFunction(x + y)
console.log(add(-1, -1, absolute)
// 20
```

This is what is being used throughout the project currently. Examples:
- [`Prompt` instance](https://github.com/eggheadio/egghead-instructor-center/blob/master/src/screens/Lessons/screens/New/components/Instructions/index.js#L5-Lundefined)
- [`Error` instance](/https://github.com/eggheadio/egghead-instructor-center/blob/master/src/screens/Lessons/screens/New/components/Propose/index.js#L123-Lundefined)
etc.